### PR TITLE
Targetting fullest storage

### PR DESCRIPTION
**Pull request by nicmictho**
closes #23

### CODE REVIEW

File: src/role/hauler.js [hunk 1]
```diff
@@ -5,12 +5,17 @@ var roleHauler = {
     run: function(creep) {
 
         if(creep.carry.energy == 0) {
-	        targetStorage = targetter.withdraw(creep);
-	        if(creep.withdraw(targetStorage, RESOURCE_ENERGY) == ERR_NOT_IN_RANGE) {
-             	creep.moveTo(targetStorage, {visualizePathStyle: {stroke: '#ffaa00'}});
+            if (!creep.memory.targetStorage){
+                console.log(targetter.withdraw(creep))
```
Comment (BWeesy) [inline src/role/hauler.js:9]: Remove logs pls

File: src/utils/spawning.js [hunk 2]
```diff
@@ -48,9 +48,10 @@ function spawnUpgrader() {
 
 function spawnHauler() {
     var bodyPieces = [MOVE, CARRY];
-    var cost = body.MOVE.cost + body.CARRY.cost;
+    var cost = body.MOVE.cost + 2 * body.CARRY.cost;
     while(cost + body.WORK.cost + body.CARRY.cost <= availableEnergy){
```
Comment (BWeesy) [inline src/utils/spawning.js:52]: Update cost in this loop to match what's in the loop pls.

File: src/utils/targetting.js [hunk 3]
```diff
@@ -62,7 +62,7 @@ var targetter = {
         });
 
         if (targets.length > 0) {
-            targetStorage = _.sortBy(targets, t => creep.pos.getRangeTo(t))[0];
+            targetStorage = _.sortBy(targets, t => t.store[RESOURCE_ENERGY])[targets.length -1];
```
Comment (BWeesy) [inline src/utils/targetting.js:65]: Please use lodash last.
https://lodash.com/docs/4.17.15#last

**Review by BWeesy — CHANGES_REQUESTED** [review]

**Comment by nicmictho** [comment]
also closes #25

**Review by BWeesy — APPROVED** [review]

**Pull request merged**